src/app/actions/shortcuts.actions.ts

```diff
@@ -1,136 +1,301 @@
+      case 'alt,c,1':
+        return [newState, fxFontColorChange(newState, 'red')];
+      case 'alt,c,2':
+        return [newState, fxFontColorChange(newState, 'blue')];
+      case 'alt,c,3':
+        return [newState, fxFontColorChange(newState, 'green')];
+      case 'alt,c,4':
+        return [newState, fxFontColorChange(newState, 'yellow')];
+      case 'alt,c,5':
+        return [newState, fxFontColorChange(newState, 'orange')];
+      case 'alt,c,6':
+        return [newState, fxFontColorChange(newState, 'purple')];
+      case 'alt,c,7':
+        return [newState, fxFontColorChange(newState, 'black')];
+      case 'alt,c,8':
+        return [newState, fxFontColorChange(newState, 'grey')];
+      case 'alt,c,9':
+        return [newState, fxFontColorChange(newState, 'white')];
+      case 'l,+':
+      case 'l,=':
+        if (!newState.fontLineSpacingActive) {
+          newState = {...newState, fontLineSpacingActive: true};
+        }
+        return [newState, fxFontLineSpacingInc(newState)];
+      case 'l,-':
+      case 'l,_':
+        if (!newState.fontLineSpacingActive) {
+          newState = {...newState, fontLineSpacingActive: true};
+        }
+        return [newState, fxFontLineSpacingDec(newState)];
+      case 'c,+':
+      case 'c,=':
+        if (!newState.fontLetterSpacingActive) {
+          newState = {...newState, fontLetterSpacingActive: true};
+        }
+        return [newState, fxFontLetterSpacingInc(newState)];
+      case 'c,-':
+      case 'c,_':
+        if (!newState.fontLetterSpacingActive) {
+          newState = {...newState, fontLetterSpacingActive: true};
+        }
+        return [newState, fxFontLetterSpacingDec(newState)];
+      case 'alt,m':
+        return [newState, fxMagShow(newState)];
+      case 'alt,o,1':
+        if (!newState.maskActive) {
+          newState = {...newState, maskActive: true};
+        }
+        return [newState, fxMaskShow(newState, 'red')];
+      case 'alt,o,2':
+        if (!newState.maskActive) {
+          newState = {...newState, maskActive: true};
+        }
+        return [newState, fxMaskShow(newState, 'blue')];
+      case 'alt,o,3':
+        if (!newState.maskActive) {
+          newState = {...newState, maskActive: true};
+        }
+        return [newState, fxMaskShow(newState, 'green')];
+      case 'alt,o,4':
+        if (!newState.maskActive) {
+          newState = {...newState, maskActive: true};
+        }
+        return [newState, fxMaskShow(newState, 'yellow')];
+      case 'alt,o,5':
+        if (!newState.maskActive) {
+          newState = {...newState, maskActive: true};
+        }
+        return [newState, fxMaskShow(newState, 'orange')];
+      case 'alt,o,6':
+        if (!newState.maskActive) {
+          newState = {...newState, maskActive: true};
+        }
+        return [newState, fxMaskShow(newState, 'purple')];
+      case 'alt,r':
+        // open reading bar
+        if (newState.rulerReadingActive) {
+          return;
```

Comment:
empty returns will crash hyperapp

```suggestion
          return newState;
```